core/src/main/java/org/apache/calcite/rex/RexExecutorImpl.java

```diff
@@ -126,25 +126,30 @@ public static RexExecutable getExecutable(RexBuilder rexBuilder, List<RexNode> e
 
   /**
    * Do constant reduction using generated code.
+   * Returns the reduced expressions in `reducedValues`, which
+   * is supposed to be empty on call.
    */
   @Override public void reduce(RexBuilder rexBuilder, List<RexNode> constExps,
       List<RexNode> reducedValues) {
-    String code;
+    // If this is not empty, the reducedValues.clear() call below is wrong.
```

Comment:
I don't know Why the java comment is `If this is not empty, the reducedValues.clear() call below is wrong` . Maybe we don't need it, use `assert reducedValues.isEmpty();` is enough.

Reply:
I guess this comment is now redundant with the JavaDoc.
Previously it wasn't obvious that reducedValues must be empty, but I checked all the callers.

Reply:
I have pushed a commit with an improved JavaDoc.